src/Invoice.cs

```diff
@@ -11,9 +11,67 @@ public class Invoice : Entity
 
         public Invoice Create(Dictionary<string, object> data)
         {
-            string relativeUrl = GetEntityUrl();
+            string relativeUrl = string.Format("{0}/{1}", GetUrlVersion(), GetEntityUrl());
             List<Entity> entities = Request(relativeUrl, HttpMethod.POST, data);
             return (Invoice)entities[0];
         }
+
+        public List<Invoice> All(Dictionary<string, object> options = null)
+        {
+            List<Entity> entities = base.All(options);
+            List<Invoice> invoices = new List<Invoice>();
+            foreach (Entity entity in entities)
+            {
+                invoices.Add(entity as Invoice);
+            }
+            return invoices;
+        }
+
+        public Invoice Edit(Dictionary<string, object> data)
+        {
+            string relativeUrl = string.Format("{0}/{1}/{2}", GetUrlVersion(), GetEntityUrl(), this["id"]);
+            List<Entity> entities = Request(relativeUrl, HttpMethod.PATCH, data);
+            return (Invoice)entities[0];
+        }
+
+        public Invoice Cancel()
+        {
+            string relativeUrl = string.Format("{0}/invoices/{1}/cancel", GetUrlVersion(), this["id"]);
+            List<Entity> entities = Request(relativeUrl, HttpMethod.POST, null);
+            return (Invoice)entities[0];
```

Comment:
Check if array element with index 0 exists before directly fetching the value.

Reply:
Add null check